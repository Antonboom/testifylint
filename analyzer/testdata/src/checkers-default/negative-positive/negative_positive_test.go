// Code generated by testifylint/internal/testgen. DO NOT EDIT.

package negativepositive

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestNegativePositiveChecker(t *testing.T) {
	var a int

	// Invalid.
	{
		assert.Greater(t, 0, a)                                   // want "negative-positive: use assert\\.Positive"
		assert.Greaterf(t, 0, a, "msg with args %d %s", 42, "42") // want "negative-positive: use assert\\.Positivef"
		assert.Greater(t, a, 0)                                   // want "negative-positive: use assert\\.Negative"
		assert.Greaterf(t, a, 0, "msg with args %d %s", 42, "42") // want "negative-positive: use assert\\.Negativef"
		assert.Less(t, 0, a)                                      // want "negative-positive: use assert\\.Negative"
		assert.Lessf(t, 0, a, "msg with args %d %s", 42, "42")    // want "negative-positive: use assert\\.Negativef"
		assert.Less(t, a, 0)                                      // want "negative-positive: use assert\\.Positive"
		assert.Lessf(t, a, 0, "msg with args %d %s", 42, "42")    // want "negative-positive: use assert\\.Positivef"
	}

	// Valid.
	{
		assert.Greater(t, 1, a)
		assert.Greaterf(t, 1, a, "msg with args %d %s", 42, "42")
		assert.Less(t, 1, a)
		assert.Lessf(t, 1, a, "msg with args %d %s", 42, "42")
	}

	// Ignored.
	{
		assert.Greater(t, 1, a)
		assert.Greaterf(t, 1, a, "msg with args %d %s", 42, "42")
		assert.Greater(t, -1, a)
		assert.Greaterf(t, -1, a, "msg with args %d %s", 42, "42")
		assert.Greater(t, 0, 0)
		assert.Greaterf(t, 0, 0, "msg with args %d %s", 42, "42")
		assert.GreaterOrEqual(t, 0, a)
		assert.GreaterOrEqualf(t, 0, a, "msg with args %d %s", 42, "42")
		assert.GreaterOrEqual(t, a, 0)
		assert.GreaterOrEqualf(t, a, 0, "msg with args %d %s", 42, "42")
		assert.Less(t, 1, a)
		assert.Lessf(t, 1, a, "msg with args %d %s", 42, "42")
		assert.Less(t, -1, a)
		assert.Lessf(t, -1, a, "msg with args %d %s", 42, "42")
		assert.Less(t, 0, 0)
		assert.Lessf(t, 0, 0, "msg with args %d %s", 42, "42")
		assert.LessOrEqual(t, 0, a)
		assert.LessOrEqualf(t, 0, a, "msg with args %d %s", 42, "42")
		assert.LessOrEqual(t, a, 0)
		assert.LessOrEqualf(t, a, 0, "msg with args %d %s", 42, "42")
	}
}
