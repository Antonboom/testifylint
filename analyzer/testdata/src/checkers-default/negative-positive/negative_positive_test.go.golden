// Code generated by testifylint/internal/testgen. DO NOT EDIT.

package negativepositive

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestNegativePositiveChecker(t *testing.T) {
	var a int

	// Invalid.
	{
		assert.Positive(t, a)                                   // want "negative-positive: use assert\\.Positive"
		assert.Positivef(t, a, "msg with args %d %s", 42, "42") // want "negative-positive: use assert\\.Positivef"
		assert.Negative(t, a)                                   // want "negative-positive: use assert\\.Negative"
		assert.Negativef(t, a, "msg with args %d %s", 42, "42") // want "negative-positive: use assert\\.Negativef"
		assert.Negative(t, a)                                   // want "negative-positive: use assert\\.Negative"
		assert.Negativef(t, a, "msg with args %d %s", 42, "42") // want "negative-positive: use assert\\.Negativef"
		assert.Positive(t, a)                                   // want "negative-positive: use assert\\.Positive"
		assert.Positivef(t, a, "msg with args %d %s", 42, "42") // want "negative-positive: use assert\\.Positivef"
	}

	// Valid.
	{
		assert.Greater(t, 1, a)
		assert.Greaterf(t, 1, a, "msg with args %d %s", 42, "42")
		assert.Less(t, 1, a)
		assert.Lessf(t, 1, a, "msg with args %d %s", 42, "42")
	}

	// Ignored.
	{
		assert.Greater(t, 1, a)
		assert.Greaterf(t, 1, a, "msg with args %d %s", 42, "42")
		assert.Greater(t, -1, a)
		assert.Greaterf(t, -1, a, "msg with args %d %s", 42, "42")
		assert.Greater(t, 0, 0)
		assert.Greaterf(t, 0, 0, "msg with args %d %s", 42, "42")
		assert.GreaterOrEqual(t, 0, a)
		assert.GreaterOrEqualf(t, 0, a, "msg with args %d %s", 42, "42")
		assert.GreaterOrEqual(t, a, 0)
		assert.GreaterOrEqualf(t, a, 0, "msg with args %d %s", 42, "42")
		assert.Less(t, 1, a)
		assert.Lessf(t, 1, a, "msg with args %d %s", 42, "42")
		assert.Less(t, -1, a)
		assert.Lessf(t, -1, a, "msg with args %d %s", 42, "42")
		assert.Less(t, 0, 0)
		assert.Lessf(t, 0, 0, "msg with args %d %s", 42, "42")
		assert.LessOrEqual(t, 0, a)
		assert.LessOrEqualf(t, 0, a, "msg with args %d %s", 42, "42")
		assert.LessOrEqual(t, a, 0)
		assert.LessOrEqualf(t, a, 0, "msg with args %d %s", 42, "42")
	}
}
